
syntax = "proto3";

package kart.lights;

// Light component IDs
enum LightComponentId {
  FRONT = 0;
  REAR = 1;
  INTERIOR = 2;
  AUXILIARY = 3;
  UNDERGLOW = 4;
}

// Light command IDs
enum LightCommandId {
  MODE = 0;
  SIGNAL = 1; // turn signals
  DURATION = 3;
  BRAKE = 4;
}

// Light mode values
enum LightModeValue {
  OFF = 0;
  ON = 1;
  // uses sensors to determine if it should be on or off
  AUTO = 2;
  DIM = 3;
  BRIGHT = 4;
  HAZARD = 8;
}

enum LightSignalValue {
    NONE = 0;
    LEFT = 1;
    RIGHT = 2;
}